// Copyright 2018 The Chromium Authors. All rights reserved.
// Use of this source code is govered by a BSD-style
// license that can be found in the LICENSE file or at
// https://developers.google.com/open-source/licenses/bsd

// This file defines protobufs for issues and related business
// objects, e.g., field values, comments, and attachments.

syntax = "proto3";

package monorail;

import "api/api_proto/common.proto";


// Next available tag: 4
message FieldValue {
  FieldRef field_ref = 1;
  string value = 2;
  bool is_derived = 3;
}


// Next available tag: 21
message Issue {
  string project_name = 1;
  uint32 local_id = 2;
  string summary = 3;
  StatusRef status_ref = 4;
  UserRef owner_ref = 5;
  repeated UserRef cc_refs = 6;
  repeated LabelRef label_refs = 7;
  repeated ComponentRef component_refs = 8;
  repeated IssueRef blocked_on_issue_refs = 9;
  repeated IssueRef blocking_issue_refs = 10;
  IssueRef merged_into_issue_ref = 11;
  repeated FieldValue field_values = 12;
  bool is_deleted = 13;
  // TODO(jrobbins): Phases and approvals.
  UserRef reporter_ref = 14;
  fixed32 opened_timestamp = 15;
  fixed32 closed_timestamp = 16;
  fixed32 modified_timestamp = 17;
  uint32 star_count = 18;
  bool is_spam = 19;
  uint32 attachment_count = 20;
}
